Suggested resource example to MI (`managed_identities`)  interface for Terraform

MariusStorhaug opened the pull request:
<!-- Thank you for submitting a Pull Request. Please fill out the template below.-->
# Overview/Summary

## This PR fixes/adds/changes/removes

- Adds a implementation example for the terraform interface `managed_identity`
- Fixes: #425

Credit: Vlad

### Breaking Changes

N/A

## As part of this Pull Request I have

- [x] Read the Contribution Guide and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Verified-Modules/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Verified-Modules/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Verified-Modules/)
- [x] Ensured PR tests are passing
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

matt-FFFFFF commented:
This can be simplified by adding `nullable = false` to the variable. Then we can remove the null check expression.

kewalaka commented:
This is what I've done in other AVM resources;

```terraform
  dynamic "identity" {
    for_each = var.managed_identities == {} ? [] : [var.managed_identities]
    content {
      type         = identity.value.system_assigned && length(identity.value.user_assigned_resource_ids) > 0 ? "SystemAssigned, UserAssigned" : length(identity.value.user_assigned_resource_ids) > 0 ? "UserAssigned" : "SystemAssigned"
      identity_ids = identity.value.user_assigned_resource_ids
    }
  }
```

...simplifies the for_each more, seems to work, happy to be pointed out what is wrong with it, though, fits with the existing AVM spec;

```terraform
variable "managed_identities" {
  type = object({
    system_assigned            = optional(bool, false)
    user_assigned_resource_ids = optional(set(string), [])
  })
  default = {}
}
```

MariusStorhaug commented:
@matt-FFFFFF: I updated suggestion with your feedback and added some description (aligned with the other interface types). Also formatted the suggestions with avmfix to be sure it matches the spec.
